cordapp/README.md

```diff
@@ -1,13 +1,24 @@
 # CorDapp Sample - Java
 
-This is a simple CorDapp using the Conclave API. It is licensed under the Apache 2 license, and therefore you 
-may copy/paste it to act as the basis of your own commercial or open source apps.
+This is a simple [CorDapp](https://docs.r3.com/en/platform/corda/4.8/open-source/cordapp-overview.html) using the Conclave API. It is licensed under the Apache 2 license. So, you 
+can copy/paste it to act as the basis of your commercial or open source apps.
 
-## Usage
+This CorDapp sample builds on the [hello-world](writing-hello-world.md) sample. It reverses a string using two [nodes](https://docs.r3.com/en/platform/corda/4.8/enterprise/node/component-topology.html). One of these nodes is used to load the enclave.
 
-Unlike most CorDapps this one will *only run on Linux* due to the need for an enclave. But don't fear: you can use 
-virtualisation to run it on Windows and macOS as well. For Windows and macOS, this will unfortunately require you to
-do some manual work to set up a docker container (instructions below).
+This sample does not use smart contracts. It requires only flows.
+
+The sample divides the code into several parts. You can copy/paste the packages that _don't_ have `samples` in the name.
+The code expects a DCAP-capable host. If you want to use EPID, you can edit the enclave startup code to make it use your
+Intel API keys.
+
+!!! important
+To understand this tutorial, you should read both the [Conclave Hello World tutorial](writing-hello-world.md)
+and [the Corda tutorials](https://docs.corda.net/docs/corda-os/4.7/hello-world-introduction.html).
```

Comment:
Here we link to Corda 4.7. We do use Corda 4.8 in this sample. But I haven't found tutorials in Corda 4.8 docsite! So probably good to leave it as is.